File: solutions/0203.remove-linked-list-elements/remove-linked-list-elements.py
# Definition for singly-linked list.
from typing import Optional


class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        temp = head
        if temp is None:
            return
        while (temp != None) and (temp.val == val):
            head = head.next
            temp = temp.next
        if temp is not None:
            while temp.next != None:
                if temp.next.val == val:
                    temp.next = temp.next.next
                else:
                    temp = temp.next
        return head

File: solutions/0203.remove-linked-list-elements/remove-linked-list-elements.js
/**
 * 203. Remove Linked List Elements
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function (head, val) {
	if (head !== null) {
		var n = head;
		while (n !== null && n.val == val) {
			head = head.next;
			n = n.next;
		}
		if (n !== null) {
			while (n.next !== null) {
				if (n.next.val == val) {
					n.next = n.next.next;
				} else n = n.next;
			}
		}
	}
	return head;
};
